tests/estimator_power_request_test.py

```diff
@@ -32,12 +32,16 @@ def generate_request(
 
 
 def process(client, energy_source):
-    request_json = generate_request(trainer_names[0], 2, output_type="AbsPower", energy_source=energy_source)
-    res = client.make_request(request_json)
-    res_json = json.loads(res)
-    print(res_json)
-    assert res_json["msg"] == "", "response error: {}".format(res_json["msg"])
-    assert len(res_json["powers"]) > 0, "zero powers"
+    for trainer_name in trainer_names:
+        request_json = generate_request(trainer_name, 2, output_type="AbsPower", energy_source=energy_source)
+        res = client.make_request(request_json)
+        res_json = json.loads(res)
+        print(res_json)
+        # FIXME: when using LogarithmicRegressionTrainer_0, the response is None:
+        # {"trainer_name": "LogarithmicRegressionTrainer_0", "metrics": ["bpf_cpu_time_ms", "bpf_page_cache_hit"], "system_features": ["node_info", "cpu_scaling_frequency_hertz"], "system_values": ["1", "1GHz"], "values": [[1.0, 1.0], [1.0, 1.0]], "output_type": "AbsPower", "source": "acpi"}
```

Comment:
@sunya-ch can you take a look at this error? I think the model server doesn't have `predict` for log regressor

Reply:
@rootfs  
It depends on which model db is connected to the estimator. If we want to test all available models (trainers), we must first make sure that the model exists at the estimator.
For curvefit-class models, it is possible that it cannot find fitting model parameters regarding the data input. 
I define SGDRegressor here since it is the simplest trainer that should always exist. 

If you run estimator standalone, check the configuration file.
If you run estimator with model server, check the db that connected to the model-server.

Reply:
indeed! with a more realistic feature values, the predicator now works

```
tests/estimator_power_request_test.py::test_estimator_power_request {"trainer_name": "SGDRegressorTrainer_0", "metrics": ["bpf_cpu_time_ms", "bpf_page_cache_hit"], "system_features": ["node_info", "cpu_scaling_frequency_hertz"], "system_values": ["1", "1GHz"], "values": [[1000.0, 1000.0], [1000.0, 1000.0]], "output_type": "AbsPower", "source": "acpi"}
{'powers': {'platform': [79.01245846529595, 79.01245846529595]}, 'msg': '', 'core_ratio': 4.0}
{"trainer_name": "LogarithmicRegressionTrainer_0.json", "metrics": ["bpf_cpu_time_ms", "bpf_page_cache_hit"], "system_features": ["node_info", "cpu_scaling_frequency_hertz"], "system_values": ["1", "1GHz"], "values": [[1000.0, 1000.0], [1000.0, 1000.0]], "output_type": "AbsPower", "source": "acpi"}
{'powers': {'platform': [76.55652313028382, 76.55652313028382]}, 'msg': '', 'core_ratio': 1}
{"trainer_name": "SGDRegressorTrainer_0", "metrics": ["bpf_cpu_time_ms", "bpf_page_cache_hit"], "system_features": ["node_info", "cpu_scaling_frequency_hertz"], "system_values": ["1", "1GHz"], "values": [[1000.0, 1000.0], [1000.0, 1000.0]], "output_type": "AbsPower", "source": "rapl-sysfs"}
{'powers': {'package': [146.99476788632222, 146.99476788632222], 'core': [0.0, 0.0], 'uncore': [0.0, 0.0], 'dram': [18.704254540352803, 18.704254540352803]}, 'msg': '', 'core_ratio': 0.16666666666666666}
{"trainer_name": "LogarithmicRegressionTrainer_0.json", "metrics": ["bpf_cpu_time_ms", "bpf_page_cache_hit"], "system_features": ["node_info", "cpu_scaling_frequency_hertz"], "system_values": ["1", "1GHz"], "values": [[1000.0, 1000.0], [1000.0, 1000.0]], "output_type": "AbsPower", "source": "rapl-sysfs"}
{'powers': {'package': [59.31611863686947, 59.31611863686947], 'core': [0.0, 0.0], 'uncore': [0.0, 0.0], 'dram': [14.265734715496643, 14.265734715496643]}, 'msg': '', 'core_ratio': 1}
```